tests/test-so3-smooth.cpp

```diff
@@ -73,7 +73,7 @@ BOOST_AUTO_TEST_CASE(from_quat_and_time_constructor) {
 BOOST_AUTO_TEST_CASE(from_quat_and_time_generate) {
   // Create some start variables.
   quaternion_t init_quat = quaternion_t::UnitRandom();
-  matrix3_t init_rot = init_quat.toRotationMatrix();
+  matrix3_t init_rot = init_quat.normalized().toRotationMatrix();
```

Comment:
`UnitRandom()` does not provide a unitary quaternion ?

Reply:
It should but `SO3Smooth::generate` renormalize the quaternion :

```cpp
  void generate(const quaternion_t& init_rot, const quaternion_t& end_rot,
                const time_t t_min, const time_t t_max) {
    quaternion_t init_rot_normalized = init_rot.normalized();
    quaternion_t end_rot_normalized = end_rot.normalized();
    generate(init_rot_normalized.toRotationMatrix(),
             end_rot_normalized.toRotationMatrix(), t_min, t_max);
  }
```

I think this normalization change a little bit the quaternion value and make this check fail :

```cpp
BOOST_CHECK_EQUAL(traj.get_init_rotation(), init_rot);
```